File: taskflow-dotnet-api/src/TaskFlow.Application.Contracts/TodoItems/TodoItemResponseDto.cs
using TaskFlow.Application.Contracts.BaseEntities;
using TaskFlow.Application.Contracts.Categories;
using TaskFlow.Application.Contracts.Profiles;
using TaskFlow.Domain.Shared.TodoItems;

namespace TaskFlow.Application.Contracts.TodoItems;

public class TodoItemResponseDto : EntityDto<Guid>
{
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public TodoStatus Status { get; set; }
    public TodoPriority Priority { get; set; }
    public DateTime? DueDate { get; set; }
    public bool IsArchived { get; set; }
    public DateTime? ArchivedTime { get; set; }
    
    public Guid CategoryId { get; set; }
    public CategoryResponseDto? Category { get; set; }
    
    public Guid UserId { get; set; }
}

File: taskflow-react-client/src/features/todoItems/types/todoItemResponseDto.ts
import type { CategoryResponseDto } from "../../categories/types/categoryResponseDto";

export type TodoStatus = "Backlog" | "InProgress" | "Blocked" | "Completed";
export type TodoPriority = "Low" | "Medium" | "High";

export interface TodoItemResponseDto {
  id: string;
  title: string;
  description?: string;
  status: TodoStatus;
  priority: TodoPriority;
  dueDate?: string;
  isArchived: boolean;
  archivedTime?: string;
  categoryId: string;
  category?: CategoryResponseDto;
  userId: string;
}
